builtin/providers/aws/resource_aws_launch_configuration.go

```diff
@@ -94,15 +96,26 @@ func resourceAwsLaunchConfiguration() *schema.Resource {
 func resourceAwsLaunchConfigurationCreate(d *schema.ResourceData, meta interface{}) error {
 	autoscalingconn := meta.(*AWSClient).autoscalingconn
 
-	var createLaunchConfigurationOpts autoscaling.CreateLaunchConfiguration
-	createLaunchConfigurationOpts.Name = d.Get("name").(string)
-	createLaunchConfigurationOpts.IamInstanceProfile = d.Get("iam_instance_profile").(string)
-	createLaunchConfigurationOpts.ImageId = d.Get("image_id").(string)
-	createLaunchConfigurationOpts.InstanceType = d.Get("instance_type").(string)
-	createLaunchConfigurationOpts.KeyName = d.Get("key_name").(string)
-	createLaunchConfigurationOpts.UserData = d.Get("user_data").(string)
-	createLaunchConfigurationOpts.AssociatePublicIpAddress = d.Get("associate_public_ip_address").(bool)
-	createLaunchConfigurationOpts.SpotPrice = d.Get("spot_price").(string)
+	var createLaunchConfigurationOpts autoscaling.CreateLaunchConfigurationType
+	createLaunchConfigurationOpts.LaunchConfigurationName = aws.String(d.Get("name").(string))
+	createLaunchConfigurationOpts.ImageID = aws.String(d.Get("image_id").(string))
+	createLaunchConfigurationOpts.InstanceType = aws.String(d.Get("instance_type").(string))
+
+	if v, ok := d.GetOk("user_data"); ok {
+		createLaunchConfigurationOpts.UserData = aws.String(base64.StdEncoding.EncodeToString([]byte(v.(string))))
+	}
+	if v, ok := d.GetOk("associate_public_ip_address"); ok {
+		createLaunchConfigurationOpts.AssociatePublicIPAddress = aws.Boolean(v.(bool))
+	}
+	if v, ok := d.GetOk("iam_instance_profile"); ok {
+		createLaunchConfigurationOpts.IAMInstanceProfile = aws.String(v.(string))
+	}
+	if v, ok := d.GetOk("key_name"); ok {
+		createLaunchConfigurationOpts.KeyName = aws.String(v.(string))
+	}
+	if v, ok := d.GetOk("spot_price"); ok {
+		createLaunchConfigurationOpts.SpotPrice = aws.String(v.(string))
+	}
```

Comment:
Yeah goamz used to do all these checks for us, it makes sense that you have to conditionalize this now.